src/v/cluster/cluster_discovery.cc

```diff
@@ -185,4 +366,37 @@ cluster_discovery::get_node_index_in_seed_servers() {
       boost::numeric_cast<int32_t>(std::distance(seed_servers.cbegin(), it))};
 }
 
+bool cluster_discovery::is_amnesiac_seed_server() const {
```

Comment:
nit: maybe more obvious what this means if called something like `seeds_already_have_cluster_uuid` or `remote_cluster_uuid_exists` or somesuch

It seems possible to get here by configuration only, without "wiping out" a node. E.g. start a new node configured as a seed, but expect it to join an existing cluster